src/Database/LSMTree/Internal/Run.hs

```diff
@@ -30,4 +35,127 @@
 -- The above list is a sketch. Functionality may move around, and the list is
 -- not exhaustive.
 --
-module Database.LSMTree.Internal.Run () where
+module Database.LSMTree.Internal.Run (
+    RunLocation (..)
+  , runKOpsPath
+  , runFilterPath
+  , runIndexPath
+  , runBlobPath
+
+  , Run
+  , fromWriteBuffer
+  ) where
+
+import qualified Control.Monad.ST as ST
+import qualified Data.ByteString.Builder as BS
+import qualified Data.ByteString.Short as BS
+import           Data.Foldable (for_)
+import           Database.LSMTree.Common (SomeSerialisationConstraint (..))
+import           Database.LSMTree.Internal.Entry
+import           Database.LSMTree.Internal.Run.BloomFilter (Bloom)
+import           Database.LSMTree.Internal.Run.Construction (BlobRef, RawValue)
+import qualified Database.LSMTree.Internal.Run.Construction as Cons
+import           Database.LSMTree.Internal.Run.Index.Compact (CompactIndex)
+import           Database.LSMTree.Internal.Serialise (SerialisedKey)
+import qualified Database.LSMTree.Internal.Serialise as S
+import           Database.LSMTree.Internal.WriteBuffer (WriteBuffer)
+import qualified Database.LSMTree.Internal.WriteBuffer as WB
+import           System.FilePath ((</>))
+import qualified System.IO as IO
+
+-- TODO: makes sense? in any case, it should live somewhere else
```

Comment:
Yes it makes sense, and indeed we do need to remember the names of the files, not just remember the open file handles, because:
1. when we close the last reference to a run then we need to delete the files; and
2. we need the names to hard link the files into a snapshot

We should be able to use this type in both `Run` and `MRun`.